src/backend/distributed/planner/shard_pruning.c

```diff
@@ -578,25 +568,104 @@ PrunableExpressionsWalker(Node *node, ClauseWalkerContext *context)
 
 		return false;
 	}
-	else
+
+	return expression_tree_walker(node, PrunableExpressionsWalker, context);
+}
+
+
+/*
+ * AddSAOPartitionKeyRestrictionToInstance adds partcol = arrayelem operator
+ * restriction to the current pruning instance.
+ */
```

Comment:
This description seems inaccurate, we're creating a new pruning instance for every element.
I'd also appreciate some more context on why we want to do this.